pkg/binding/bindingpolicy-resolver.go

```diff
@@ -345,6 +350,21 @@ func (resolver *bindingPolicyResolver) GetSingletonReportedStateRequestForObject
 	return requested, matchingWECs.Len()
 }
 
+func (resolver *bindingPolicyResolver) GetSingletonReportedStateRequestsForBinding(bindingPolicyKey string) ([]util.ObjectIdentifier, []bool, []int) {
```

Comment:
@MikeSpreitzer @waltforme why is this not a part of a new subscriber to the broker?
I don't think we want to bloat this interface.

This way the status can return to be a user of the broker, and not the resolver - which is a weaker dependency.

Apologies for the late review.

Reply:
Can you pleae elaborate on the alternative that you are suggesting? I am not sure I understand exactly what you mean by "this" being "the responsibility of a new subscriber to the broker".

Reply:
This PR switched the status controller from being a user of the Broker to a user of the Resolver, which is a stronger and unnecessary dependency imo.

All the new Resolver function does is iterate over objects of a resolution given a bindingKey, which can easily be implemented in a new status controller sub-structure that uses the Broker. This way, the status remains a user of the Broker only (which is sort of an informer), and the responsibility of calculating this information is better placed - instead of bloating the binding Resolver.

Reply:
Let me see if I have got it right. `Broker` is essentially an informer of resolutions and the status controller should remain a user of only that (possibly indirectly through the new thing mentioned in the following sentence). The function of iterating over the objects of a resolution given a bindingKey should be a method of a new thing that only uses the Broker. Have I got that right?

Would the new thing that only uses the broker go in its own new file, or be added to `bindingpolicy-resolution-broker.go`?

Reply:
@vMaroon: are you suggesting that `GetSingletonReportedStateRequestsForBinding` could depend only on one resolution rather than everything that a resolver holds? I do not follow that. `GetSingletonReportedStateRequestsForBinding` needs access to all the resolutions in order to do its job properly, which includes finding whether any resolution requests singleton state return and counting the number of all the associated WECs.

Reply:
@MikeSpreitzer on the first message, yes. A new structure should reside in the status package and use the broker. On the latter, no, I'm not saying it can depend on a single resolution. I'm saying the broker as-is can provide everything needed in that function.

Right now, the consumer would need to keep track of existing bindings since the broker lacks a method that provides the list or a lazy iterator.

The end point is to keep the binding resolver compact, and have derived logic be built on using the broker.